src/status_im/contexts/wallet/events.cljs

```diff
@@ -128,30 +136,36 @@
                                     :event :wallet/remove-account})}]]]}))
 
 (rf/reg-event-fx
- :wallet/get-wallet-token
+ :wallet/get-wallet-token-for-all-accounts
  (fn [{:keys [db]}]
-   (let [addresses (->> (get-in db [:wallet :accounts])
-                        vals
-                        (map :address))]
-     {:db (assoc-in db [:wallet :ui :tokens-loading?] true)
-      :fx [[:json-rpc/call
-            [{:method     "wallet_getWalletToken"
-              :params     [addresses]
-              :on-success [:wallet/store-wallet-token]
-              :on-error   [:wallet/get-wallet-token-failed addresses]}]]]})))
+   {:fx (->> (get-in db [:wallet :accounts])
+             vals
+             (mapv
+              (fn [{:keys [address]}]
+                [:dispatch [:wallet/get-wallet-token-for-account address]])))}))
```

Comment:
Do we have tests for these events already? If not, can we add some for the refactored code?

Reply:
No, we need to write tests for wallet events 😞. I added a few for the events I have modified. But, we need to write test all events.